internal/balancer/gracefulswitch/gracefulswitch.go

```diff
@@ -76,7 +78,9 @@ func (gsb *Balancer) swap() {
 	cur := gsb.balancerCurrent
 	gsb.balancerCurrent = gsb.balancerPending
 	gsb.balancerPending = nil
+	gsb.pendingSwaps.Add(1)
 	go func() {
+		defer gsb.pendingSwaps.Done()
```

Comment:
How about using the newly added `Go` method on the `sync.WaitGroup` to run goroutines on it?